### DISCUSSION TITLE
View Submissions

### DISCUSSION THREAD

**Discussion by javawolfpack** [INGInious/INGInious#948] [Q&A] [answered]
I'm getting the following error page when I try to look at submissions:

<img width="1139" alt="Screenshot 2023-05-23 at 9 47 27 AM" src="https://github.com/UCL-INGI/INGInious/assets/1179047/486c5304-6f8c-4be5-b0d1-fbcf7e0af06d">

This is after solving the empty container issue resolved here: https://github.com/UCL-INGI/INGInious/issues/650

Not sure if there is something else missing here? Does the view submissions in the webapp require the webdav to be working? Per issue https://github.com/UCL-INGI/INGInious/issues/944 I am working on getting a sub-subdomain instead of routing to a different path, it still loads and prompts me for a password, just not authenticating still.

**Comment by javawolfpack** [comment]
So I got sub-subdomain mapped, just entered admin w/ no password, and that somehow progressed when my user account didn't. <img width="653" alt="Screenshot 2023-05-23 at 10 04 46 AM" src="https://github.com/UCL-INGI/INGInious/assets/1179047/372028a5-f6ef-4b9b-9391-e7357070f297">

And this did require me manually go to the above address path as just `https://webdav.inginious.csuchico.edu` fails to load. Get the following 404:

```
404 Not Found: Could not resolve realm for /
```

Have this in my `configuration.yaml` is there something missing? 

```yaml
webdav_host: https://webdav.inginious.csuchico.edu
```

**Reply by anthonygego** [reply]
Please check what it does with the info provided in top a the course task list.

**Reply by javawolfpack** [reply]
Okay resolved the webdav issue as your reply https://github.com/UCL-INGI/INGInious/issues/944#issuecomment-1560575979 made me realize it uses different credentials generated on the page.

**Comment by anthonygego** [comment]
> I'm getting the following error page when I try to look at submissions:
> 
> <img alt="Screenshot 2023-05-23 at 9 47 27 AM" width="1139" src="https://user-images.githubusercontent.com/1179047/240363999-486c5304-6f8c-4be5-b0d1-fbcf7e0af06d.png">
> This is after solving the empty container issue resolved here: #650
> 
> Not sure if there is something else missing here? Does the view submissions in the webapp require the webdav to be working? Per issue #944 I am working on getting a sub-subdomain instead of routing to a different path, it still loads and prompts me for a password, just not authenticating still.

Never seen such a verbose message and have no idea what part of the stack generates it (probably flask ?). There should be something more interesting in the webapp logs can help to know what's the actual issue. Probably a stack trace or at least something more specific.

**Reply by javawolfpack** [reply]
Yep, looks like a KeyError for `superadmin`

```syslog
May 24 17:05:42 inginious uwsgi[115907]: 2023-05-24 17:05:42,549 - inginious.frontend.app - ERROR - Exception on /admin/tutorial/submissions [GET]
May 24 17:05:42 inginious uwsgi[115907]: Traceback (most recent call last):
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/flask/app.py", line 2190, in wsgi_app
May 24 17:05:42 inginious uwsgi[115907]:     response = self.full_dispatch_request()
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/flask/app.py", line 1486, in full_dispatch_request
May 24 17:05:42 inginious uwsgi[115907]:     rv = self.handle_user_exception(e)
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/flask/app.py", line 1484, in full_dispatch_request
May 24 17:05:42 inginious uwsgi[115907]:     rv = self.dispatch_request()
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/flask/app.py", line 1469, in dispatch_request
May 24 17:05:42 inginious uwsgi[115907]:     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/flask/views.py", line 109, in view
May 24 17:05:42 inginious uwsgi[115907]:     return current_app.ensure_sync(self.dispatch_request)(**kwargs)
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/flask/views.py", line 190, in dispatch_request
May 24 17:05:42 inginious uwsgi[115907]:     return current_app.ensure_sync(meth)(**kwargs)
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/inginious/frontend/pages/utils.py", line 62, in get
May 24 17:05:42 inginious uwsgi[115907]:     return pre_check if pre_check else self.GET(*args, **kwargs)
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/inginious/frontend/pages/utils.py", line 186, in GET
May 24 17:05:42 inginious uwsgi[115907]:     return self.GET_AUTH(*args, **kwargs)
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/inginious/frontend/pages/course_admin/submissions.py", line 111, in GET_AUTH
May 24 17:05:42 inginious uwsgi[115907]:     params = self.get_input_params(user_input, course)
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/inginious/frontend/pages/course_admin/utils.py", line 81, in get_input_params
May 24 17:05:42 inginious uwsgi[115907]:     users = self.get_users(course)
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/inginious/frontend/pages/course_admin/utils.py", line 77, in get_users
May 24 17:05:42 inginious uwsgi[115907]:     users = {user: users_info[user].realname if users_info[user] else '' for user in user_ids}
May 24 17:05:42 inginious uwsgi[115907]:   File "/usr/local/lib/python3.10/dist-packages/inginious/frontend/pages/course_admin/utils.py", line 77, in <dictcomp>
May 24 17:05:42 inginious uwsgi[115907]:     users = {user: users_info[user].realname if users_info[user] else '' for user in user_ids}
May 24 17:05:42 inginious uwsgi[115907]: KeyError: 'superadmin'
```

The account I'm logged in is listed as a superadmin, though the config/documentation list it as plural:

```yaml
superadmins:
  - bryan
```

**Reply by anthonygego** [reply, marked as answer]
According to the piece of code that fails, you may have `superadmin` user registered to the course but not present in the `users` DB collection. This should theoretically not happen. Did you delete the `superadmin` user or re-configured your instance on an existing DB ?

To solve this you can probably just modify the or delete (at this setup stage, probably harmless) the entry in DB using mongosh: ``db.courses.delete_one({"_id": "tutorial"})``. With chances you can also do it from the GUI user management page because it may still be (historically) failsafe regarding the user database information. 

However that would be useful if you recall what was done using the superadmin account up until now to better identify if there is a bug to solve or if it was a mishandling.

**Reply by javawolfpack** [reply]
I ran into an issue when I first installed regarding not allowing duplicate emails... so re-ran the installer and made a new superadmin with an email that wouldn't match my SAML2 user eventually. I had a vmware snapshot prior to doing that so just backed up all my configs, certs, etc and reinstalled. Can see submissions now.

Thanks for the help on this!!
